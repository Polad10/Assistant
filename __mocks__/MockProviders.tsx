import { Dispatch, PropsWithChildren, SetStateAction } from 'react'
import { Translator } from '../helpers/Translator'
import { AppointmentRequest } from '../modals/Appointment'
import { PatientRequest } from '../modals/Patient'
import { Setting } from '../modals/Setting'
import { TreatmentRequest } from '../modals/Treatment'
import { PaymentRequest } from '../modals/Payment'
import { Api } from '../helpers/Api'
import { mockModels } from './MockModels'
import { AuthContext } from '../contexts/AuthContext'
import { DataContext } from '../contexts/DataContext'
import { ThemeContext } from '../contexts/ThemeContext'
import { LocalizationContext } from '../contexts/LocalizationContext'

const themeContextMock = {
  primary: '#121212',
  secondary: '#1a1a1a',
  accent: '#7600eb',
  neutral: '#faf8ff',
  border: '#374151',
  info: '#d3fbd8',
  success: '#00e5b0',
  warning: '#9f1239',
  error: '#9f1239',
}

const localizationContextMock = {
  translator: new Translator('en'),
  language: 'en',
}

const authContextMock = {
  user: undefined,
  loading: false,
}

const dataContextMock = {
  fetchPatients: async () => {},
  fetchAppointments: async () => {},
  fetchTreatments: async () => {},
  fetchPayments: async () => {},
  fetchSetting: async () => {},

  createAppointment: async (appointmentRequest: AppointmentRequest) => mockModels.appointment_1,
  updateAppointment: async (appointmentRequest: AppointmentRequest) => mockModels.appointment_1,
  deleteAppointment: async (appointmentId: number) => {},

  createPatient: async (patientRequest: PatientRequest) => mockModels.patient_1,
  updatePatient: async (patientRequest: PatientRequest) => mockModels.patient_1,
  deletePatient: async (patientId: number) => {},

  createTreatment: async (treatmentRequest: TreatmentRequest) => mockModels.treatment_1,
  updateTreatment: async (treatmentRequest: TreatmentRequest) => mockModels.treatment_1,
  deleteTreatment: async (treatmentId: number) => {},

  createPayment: async (paymentRequest: PaymentRequest) => mockModels.payment_1,
  updatePayment: async (paymentRequest: PaymentRequest) => mockModels.payment_1,
  deletePayment: async (paymentId: number) => {},

  createSetting: async (setting: Setting) => {},
  updateSetting: async (setting: Setting) => {},

  api: undefined,
  setApi: (action: SetStateAction<Api | undefined>): Dispatch<SetStateAction<Api | undefined>> => jest.fn(),

  loading: false,
  setLoading: (action: SetStateAction<boolean>): Dispatch<SetStateAction<boolean>> => jest.fn(),

  patients: [mockModels.patient_1, mockModels.patient_2],
  appointments: [mockModels.appointment_1, mockModels.appointment_2],
  treatments: [mockModels.treatment_1, mockModels.treatment_2],
  payments: [mockModels.payment_1, mockModels.payment_2],
  setting: mockModels.setting,
}

export default function MockProviders({ children }: PropsWithChildren) {
  return (
    <AuthContext.Provider value={authContextMock}>
      <DataContext.Provider value={dataContextMock}>
        <ThemeContext.Provider value={themeContextMock}>
          <LocalizationContext.Provider value={localizationContextMock}>{children}</LocalizationContext.Provider>
        </ThemeContext.Provider>
      </DataContext.Provider>
    </AuthContext.Provider>
  )
}
