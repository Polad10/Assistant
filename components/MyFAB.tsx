import { FAB, FABProps } from '@rneui/themed'
import { StyleSheet } from 'react-native'
import { useContext } from 'react'
import { ThemeContext } from '../contexts/ThemeContext'

export default function MyFAB(props: FABProps) {
  const themeContext = useContext(ThemeContext)!

  return (
    <FAB
      {...props}
      icon={{ name: 'add', color: themeContext.neutral }}
      placement='right'
      color={themeContext.accent}
      style={styles.fab}
    />
  )
}

const styles = StyleSheet.create({
  fab: {
    marginRight: 20,
    marginBottom: 30,
  },
})
