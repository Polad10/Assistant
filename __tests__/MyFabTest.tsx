import renderer from 'react-test-renderer'
import MockProviders from '../__mocks__/MockProviders'
import MyFAB from '../components/MyFAB'

describe('MyFab', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MockProviders>
          <MyFAB />
        </MockProviders>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
