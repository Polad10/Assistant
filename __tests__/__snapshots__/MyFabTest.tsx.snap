// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyFab renders correctly 1`] = `
<View
  collapsable={false}
  style={
    {
      "alignItems": "center",
      "borderRadius": 28,
      "bottom": 0,
      "flexDirection": "row",
      "justifyContent": "center",
      "margin": 16,
      "marginBottom": 30,
      "marginRight": 20,
      "opacity": 1,
      "position": "absolute",
      "right": 0,
      "transform": [
        {
          "scale": 1,
        },
      ],
    }
  }
>
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        {
          "borderRadius": 2,
        },
        {
          "borderRadius": 28,
          "elevation": 4,
        },
        false,
      ]
    }
    testID="RNE_BUTTON_WRAPPER"
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": false,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#7600eb",
            "borderColor": "#2089dc",
            "borderRadius": 2,
            "borderWidth": 0,
            "flexDirection": "row",
            "height": 56,
            "justifyContent": "center",
            "padding": 16,
            "paddingHorizontal": 10,
            "width": 56,
          }
        }
      >
        <View
          style={
            {
              "height": 56,
              "marginHorizontal": 5,
              "overflow": "hidden",
              "padding": 16,
              "width": 56,
            }
          }
          testID="RNE__ICON__CONTAINER"
        >
          <View
            accessibilityRole="button"
            android_ripple={
              {
                "borderless": false,
                "color": "rgba(0, 0, 0, 0.3)",
                "radius": -5,
              }
            }
            disabled={false}
            testID="RNE__ICON__CONTAINER_ACTION"
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
              testID="RNE__ICON"
            >
              <Text />
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
